crates/chain/src/spk_client.rs

```diff
@@ -239,6 +239,7 @@ pub struct SyncRequest<I = ()> {
     txids_consumed: usize,
     outpoints: VecDeque<OutPoint>,
     outpoints_consumed: usize,
+    time_of_sync: u64,
```

Comment:
```suggestion
    time_of_sync: Option<u64>,
```

I think it's important that the caller can opt out of setting a seen_at for unconfirmed transactions.

Additionally, I don't think having a default of 0 is a good idea... However, I need to think more on this.